### PR TITLE
Feature/federated prometheus

### PR DESCRIPTION
**Pull request by charlie-pisuraj**
Enables federation of prometheus cluster if the user enables exposing prometheus externally

### CODE REVIEW

File: build/deploy/metadata_base.libsonnet [hunk 1]
```diff
@@ -43,4 +43,9 @@
       dest: error 'must specify destination email',
     },
   },
+  prometheus: {
+    external: false,
```
Comment (steeling) [inline build/deploy/metadata_base.libsonnet]: can we make this variable more descriptive?

File: build/deploy/metadata_base.libsonnet [hunk 2]
```diff
@@ -43,4 +43,9 @@
...
+    IP: '',  // Leaving blank means your cloud provider will assign an ephemeral IP
```
Comment (steeling) [inline build/deploy/metadata_base.libsonnet]: can you add a comment on what populating it does?

File: build/deploy/metadata_base.libsonnet [hunk 3]
```diff
@@ -43,4 +43,9 @@
...
+    whitelist_ip_ranges: [],  // Empty list means firewall rules are open
```
Comment (steeling) [inline build/deploy/metadata_base.libsonnet]: add a comment meaning anyone can access via public internet... in fact let's have an assertion somewhere that checks to see if we are making it external then this is not empty

File: build/deploy/metadata_base.libsonnet (hunk 1 above)
Comment (steeling) [reply build/deploy/metadata_base.libsonnet]: maybe, "exposed"

File: build/deploy/metadata_base.libsonnet (hunk 3 above)
Comment (charlie-pisuraj) [reply build/deploy/metadata_base.libsonnet]: So we don't use this field at all unless, you want to expose prometheus externally, there is another service attached to the pod for intra cluster communication

**Pull request merged**